consensus/src/marshal/mod.rs

```diff
@@ -64,6 +64,14 @@ pub mod ingress;
 pub use ingress::mailbox::Mailbox;
 pub mod resolver;
 
+use crate::{threshold_simplex::signing_scheme::SigningScheme, types::Epoch};
+
+/// Supplies the signing scheme the marshal should use for a given epoch.
+pub trait SigningSchemeProvider<S: SigningScheme>: Send + Sync + 'static {
+    /// Return the signing scheme that corresponds to `epoch`.
+    fn for_epoch(&self, epoch: Epoch) -> Option<S>;
+}
```

Comment:
Small nit - let's make `S` an associated type?


```suggestion
/// Supplies the signing scheme the marshal should use for a given epoch.
pub trait SigningSchemeProvider: Send + Sync + 'static {
    /// The type of [SigningScheme] yielded by the provider.
    type Scheme: SigningScheme;

    /// Return the signing scheme that corresponds to `epoch`.
    fn for_epoch(&self, epoch: Epoch) -> Option<Self::Scheme>;
}
```

Reply:
Good point, updated!